File: labor1/app/src/main/java/com/example/labor1/ProfileActivity.java
package com.example.labor1;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.Objects;

public class ProfileActivity extends AppCompatActivity {

    private static int callFromLogin = 0;
    private static int callFromRegister = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_profile);
        Bundle bundle = getIntent().getExtras();
        if (bundle != null) {
            ((TextView) this.findViewById(R.id.firstNameText)).setText(bundle.getString("firstName"));
            ((TextView) this.findViewById(R.id.lastNameText)).setText(bundle.getString("lastName"));
            ((TextView) this.findViewById(R.id.departmentText)).setText(bundle.getString("department"));
            String caller = bundle.getString("caller");
            if (Objects.equals(caller, "login")) {
                callFromLogin++;
            } else if (Objects.equals(caller, "register")) {
                callFromRegister++;
            }
        }
        ((TextView) this.findViewById(R.id.loginCallCounter)).setText(String.valueOf(callFromLogin));
        ((TextView) this.findViewById(R.id.registerCallCounter)).setText(String.valueOf(callFromRegister));
    }

    public void callSignIn(View view) {
        Intent signInActivity = new Intent(this, MainActivity.class);
        startActivity(signInActivity);
    }
}

File: labor1/app/src/main/java/com/example/labor1/ProfileActivity.kt
package com.example.labor1;

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import java.util.*

class ProfileActivity : AppCompatActivity() {

    companion object {
        @JvmStatic
        var callFromLogin = 0
        @JvmStatic
        var callFromRegister = 0
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_profile);
        val bundle = intent.extras
        if (bundle != null) {
            findViewById<TextView>(R.id.firstNameText).text = bundle.getString("firstName")
            findViewById<TextView>(R.id.lastNameText).text = bundle.getString("lastName")
            findViewById<TextView>(R.id.departmentText).text = bundle.getString("department")
            val caller = bundle.getString("caller")
            if (Objects.equals(caller, "login")) {
                callFromLogin++
            } else if (Objects.equals(caller, "register")) {
                callFromRegister++
            }
        }
        findViewById<TextView>(R.id.loginCallCounter).text = callFromLogin.toString()
        findViewById<TextView>(R.id.registerCallCounter).text = callFromRegister.toString()
    }

    fun callSignIn(view: View) {
        val signInActivity = Intent(this, MainActivity::class.java)
        startActivity(signInActivity)
    }
}
